src/bacnet/basic/object/bi.c

```diff
@@ -1022,6 +1022,151 @@ int Binary_Input_Event_Information(
         return 0; /* no active event at this index */
 }
 
+int Binary_Input_Alarm_Ack(
+    BACNET_ALARM_ACK_DATA *alarmack_data, BACNET_ERROR_CODE *error_code)
+{
+    BINARY_INPUT_DESCR *CurrentBI;
+    unsigned int object_index;
+
```

Comment:
Suggestions. Feel free to ignore

only if needed consider checking alamack_data for NULL pointer dereference.
also consider returning -3, -4, -5 etc... instead of -1 for most error situations